packages/instrumentation-amqplib/src/amqplib.ts

```diff
@@ -358,22 +371,134 @@ export class AmqplibInstrumentation extends InstrumentationBase<typeof amqp> {
                 );
             }
 
-            // calling original publish function is only storing the message in queue.
-            // it does not send it and waits for an ack, so the span duration is expected to be very short.
+            const patchedOnConfirm = function (err: any, ok: amqp.Replies.Empty) {
+                let callbackError;
+                context.with(trace.setSpan(markedContext, span), () => {
+                    try {
+                        callback?.call(this, err, ok);
```

Comment:
We should probably delete the `isConfirmChannel` context key before executing the callback. user may call `publish` from the callback which will be marked with `CC` thus not instrumented

Reply:
Interesting topic
Not sure what we can expect as the right behavior here. Saw there are [new SIG meetings for instrumentations](https://github.com/open-telemetry/community/pull/819/files) which I plan to attend, might be a good topic to add to the agenda and hear opinions. Will be happy to see you there as well if you are up to it.
What do you think about leaving it as it is for now and revisit the issue once discussed and documented in spec?

Reply:
I am okay with leaving it as is. For SIG. Well, I am in the middle of moving to another city so probably I won't have time for this, can't promise. You are more than welcome to discuss it though. Thank you!